### PR TITLE
BUG 0805631

### PR DESCRIPTION
**Pull request by cs-tushar-kanade**
https://mantis.fortinet.com/bug_view_page.php?bug_id=0805631

### CODE REVIEW

File: csv-data-management/extract_data_from_csv.py [hunk 1]
```diff
@@ -280,8 +280,11 @@ def _check_if_csv(filepath):
     # To-Do  Read CSV file encoding and then use it for reading file. use -chardet.detect
     try:
         res = sniffer.has_header(open(filepath).read(2048))
-    except UnicodeDecodeError:
-        raise ConnectorError("CSV file has unsupported encoding.Supported encoding is UTF-8")
+    except Exception as Err:
+        if "UnicodeDecodeError" in repr(Err):
+            raise ConnectorError("CSV file has unsupported encoding.Supported encoding is UTF-8")
+        else:
+            pass
```
Comment (cs-suyog-jinturkar) [inline csv-data-management/extract_data_from_csv.py:289]: why pass?

File: csv-data-management/extract_data_from_csv.py [hunk 2]
```diff
@@ -337,14 +340,14 @@ def _read_and_return_ds(filepath,params,config):
...
             df_file =  _read_file_specific_columns(filepath,columnNames,numberOfRowsToSkip)
             
-        elif isCSVWithoutHeaders: # CSV file without column header and more than one column
-            df_file =  _read_file_no_headers(filepath,numberOfRowsToSkip,noOfColumns) 
-
-        elif isSingleColumn: #CSV with single column and header    
+        elif isSingleColumn and not isCSVWithoutHeaders: #CSV with single column and header    
             df_file = _read_file_single_column(filepath,numberOfRowsToSkip)
         
         elif isSingleColumn and not isCSVWithoutHeaders: # CSV file with one  column and no header
```
Comment (cs-suyog-jinturkar) [inline csv-data-management/extract_data_from_csv.py]: These conditions can be further improved to enhance readability

Comment (cs-suyog-jinturkar) [reply csv-data-management/extract_data_from_csv.py]: There are two if-else block with "isSingleColumn and not isCSVWithoutHeaders:" can you please check if its correct

**Review by cs-suyog-jinturkar — COMMENTED** [review]
Added review comments

File: csv-data-management/extract_data_from_csv.py (hunk 1 above)
Comment (cs-tushar-kanade) [reply csv-data-management/extract_data_from_csv.py:289]: as i am checking only for UnicodeDecodeError and the bailing out. If UnicodeDecodeError is not present I re-read file to check for headers. those exceptions are handled in below

File: csv-data-management/extract_data_from_csv.py (hunk 2 above)
Comment (cs-tushar-kanade) [reply csv-data-management/extract_data_from_csv.py]: corrected typo

**Review by cs-suyog-jinturkar — APPROVED** [review]

**Pull request merged**